### PR TITLE
GET Bot Data api

### PR DESCRIPTION
**Pull request by tausifmuzaffar**
/botData
/botData/key
/rooms/key/data
/rooms/key/bot/key/data

### CODE REVIEW

File: tests/api/v1/botData/get.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+      .then(() => {
+        api.get(`/v1/rooms/${testBotData.roomId}/data`)
+        .set('Authorization', token)
+        .expect(constants.httpStatus.OK)
+        .end((err, res) => {
+          if (err) {
+            done(err);
+          }
+
+          expect(res.body.length).to.equal(ONE);
```
Comment (pallavi2209) [inline tests/api/v1/botData/get.js]: Do you intend to check the contents here as well? Or that implementation is still pending?

File: api/v1/controllers/botData.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   * GET /room/{roomID}/bot/{botID}/data
+   *
+   * Finds zero or more botData and sends them back in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  findRoomBotBotData(req, res, next) {
+    doFind(req, res, next, helper);
```
Comment (pallavi2209) [inline api/v1/controllers/botData.js]: I am confused. Do you intend to update the doFind logic for roomIds and botIds or is it already in place?

**Comment by iamigo** [comment]
will not review until tests passing

**Comment by shriramshankar** [comment]
@tausifmuzaffar Some conflicts to resolve here.

**Pull request merged**